lib/src/boilerplate_suggestors/advanced_props_and_state_class_migrator.dart

```diff
@@ -28,13 +28,19 @@ import 'boilerplate_utilities.dart';
 class AdvancedPropsAndStateClassMigrator extends GeneralizingAstVisitor
     with AstVisitingSuggestorMixin
     implements Suggestor {
+  final SemverHelper helper;
+
+  AdvancedPropsAndStateClassMigrator(this.helper);
+
   @override
   visitClassDeclaration(ClassDeclaration node) {
     super.visitClassDeclaration(node);
 
-    if (!shouldMigrateAdvancedPropsAndStateClass(node)) return;
+    if (!shouldMigrateAdvancedPropsAndStateClass(node, helper)) return;
```

Comment:
> The main thing I'm trying to prevent is passing around the same helper instance through all the suggestors and utilities, but if that seems like a good goal then I'm open to whatever implementation works best!

Passing around the instance is actually what I'd prefer, as opposed to a global variable.

Advantages:
- Eliminates the risk of different areas of the code unintentionally interfering with each other's data (including in tests)
-Make it easier to reason about where this is used, and force us to be intentional about where we pass it

Reply:
I could definitely see that, and normally I wouldn't advocate for a global variable, but my main thought is that the function which answers the question of whether a node is public or not is at the utility level, and it doesn't seem like the suggestor should need to be concerned with how it answers that question (outside of passing in the node).

Also, because this helper should receive the report for the relevant library it's running on, it seems like it makes sense for it to operate independently since any instantiation for an executable should be passing it the same report. For tests that does break down, but then the testing environment could have its own instance when needed. If you feel strongly, Greg, I think that generally global variables are bad, this is just a weird case to me 😄

Reply:
I feel pretty strongly here that moving away from a top-level variable would improve the modularity of this code, as well as the predictability of how data is passed around and used.

> it seems like it makes sense for it to operate independently since any instantiation for an executable should be passing it the same report

That's true, but may not always be true. For example, what if we wanted this utility to behave slightly differently for different suggestors? Or if we wanted to run multiple sets of suggestors on different directories concurrently? Having global variables would make it much more difficult to implement these changes. To be fair, this is the weakest argument of the ones I've presented, but it is something to keep in mind.

---

I'm happy to make a PR with those changes, especially since some work was already done to move it more toward top-level variables.

Reply:
Moving away from a global variable sounds good to me. Thanks for explaining your logic, Greg! And sorry for making ya change it in the first place Sydney 😅